.github/workflows/golang.yaml

```diff
@@ -115,12 +119,16 @@ jobs:
     env:
       GOPRIVATE: github.com/zaphiro-technologies/*
       GH_ACCESS_TOKEN: ${{ secrets.REPO_PRIVATE_READ_PAT }}
+      GH_USERNAME: chicco785
+      GH_PACKAGES_TOKEN: ${{ secrets.PACKAGE_READ_PAT }}
     steps:
       - uses: actions/checkout@v4
       - run: git config --global url.https://$GH_ACCESS_TOKEN@github.com/.insteadOf https://github.com/
       - uses: actions/setup-go@v4
         with:
           go-version-file: go.mod
+      - name: Docker Login
+        run: echo "$GH_PACKAGES_TOKEN" | docker login ghcr.io -u $GH_USERNAME --password-stdin
```

Comment:
```suggestion
      - name: Login to GitHub Container Registry
        uses: docker/login-action@v3
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.PACKAGE_READ_PAT }}
```

maybe?

Reply:
It's a legit alternative, but `github.actor` is not the right username for `PACKAGE_READ_PAT`, isn't it?

Reply:
correct, i wonder though why the default token works fine in the docker.yaml and here not. did you test `${{ secrets.GITHUB_TOKEN }}` ?

Reply:
Yes, I tested it. The difference is that docker.yaml pulls and pushes packages of the same repo it is building, while here we are pulling from other repositories.

Reply:
No, sorry, I tested only REPO_PRIVATE_READ_PAT

Reply:
let's the test secrets.GITHUB_TOKEN

this wf https://github.com/zaphiro-technologies/data-download-frontend/blob/main/.github/workflows/docker.yaml#L157

is pulling images from another private repo, the trick was on to authorise the repo to access the backend repo packages.

this can be done in the settings of the packages, see here what i did in this case:

https://github.com/orgs/zaphiro-technologies/packages/container/data-download-api/settings